drivers/adc/adc_sam0.c

```diff
@@ -231,20 +231,27 @@ static int adc_sam0_channel_setup(const struct device *dev,
 	if (channel_cfg->differential) {
 		INPUTCTRL |= ADC_INPUTCTRL_MUXNEG(channel_cfg->input_negative);
 
-#ifdef ADC_INPUTCTRL_DIFFMODE
+#if defined(ADC_INPUTCTRL_DIFFMODE)
 		INPUTCTRL |= ADC_INPUTCTRL_DIFFMODE;
-#else
+#elif defined(ADC_CTRLB_DIFFMODE)
 		adc->CTRLB.bit.DIFFMODE = 1;
-		wait_synchronization(adc);
+#elif defined(ADC_CTRLC_DIFFMODE)
+		adc->CTRLC.bit.DIFFMODE = 1;
+#else
+		return -EINVAL;
 #endif
 	} else {
 		INPUTCTRL |= ADC_INPUTCTRL_MUXNEG_GND;
 
```

Comment:
why not change `INPUTCTRL |= ADC_INPUTCTRL_MUXNEG_GND` to use ADC_INPUTCTRL_MUXNEG(value) ?

Reply:
```
INPUTCTRL |= ADC_INPUTCTRL_MUXNEG_GND;
```

would become

```
INPUTCTRL |= ADC_INPUTCTRL_MUXNEG(ADC_INPUTCTRL_MUXNEG_GND_Val);
```

... I see no particular benefit, and this isn't modified in the patch anyway. Can do if that's what you want though(?)

Reply:
I look the macro code and it doesn't clear regs, so, there is no point in use it, you are right. My concern is that, let's assume that driver set `INPUTCTRL |= ADC_INPUTCTRL_MUXNEG_AIN3` first. What happen if you add `INPUTCTRL |= ADC_INPUTCTRL_MUXNEG_GND` ?

Maybe driver should first clear all channels and then set the new one.

```c
INPUTCTRL &= ~ADC_INPUTCTRL_MUXNEG_Msk;
INPUTCTRL |= ADC_INPUTCTRL_MUXNEG_GND;
```

Reply:
Note that `INPUTCTRL` here is a local variable, not a register... I think this is fine.

https://github.com/zephyrproject-rtos/zephyr/blob/0d412c4ce75c2f2ae0a6eb538627bbd9b04278ed/drivers/adc/adc_sam0.c#L185

Reply:
@nandojve please confirm you're okay with this staying?

Reply:
> Note that INPUTCTRL here is a local variable, not a register... I think this is fine.

uhmm, upper case is not for local variables. Please, fix it!